synapse/handlers/register.py

```diff
@@ -553,42 +567,42 @@ def check_registration_ratelimit(self, address):
 
         self.ratelimiter.ratelimit(address)
 
-    def register_with_store(
+    async def register_with_store(
```

Comment:
Instead of returning an awaitable I made this async, which we decided was a clearer pattern to use.